### ISSUE TITLE
Build a wikipedia viewer search leads to 404 NOT FOUND

### ISSUE THREAD

**Issue by dimitrissnk** [freeCodeCamp/freeCodeCamp#17304]
I can't find the Lesson: _Build a Wikipedia Viewer_, so I used the new search box and I typed "Wikipedia Viewer". The [below ](https://search.freecodecamp.org/?q=wikipedia%20viwer) page open up's with three results. I click on the «Lesson: Build a Wikipedia Viewer» and I get a 404 not found error and the link [here](https://learn.freecodecamp.org/404.html)
Before getting the error above, it looks like the page is loading, with the green freecodecamp bar appearing and disappearing quite fast with white background but with a different url as you can see in the screenshots

![screenshot_11](https://user-images.githubusercontent.com/18249143/40750562-61662946-6470-11e8-9fc7-8fad9102255f.jpg)
![screenshot_2](https://user-images.githubusercontent.com/18249143/40750490-2b636a52-6470-11e8-8871-27a408f6201e.jpg)
![screenshot_3](https://user-images.githubusercontent.com/18249143/40750491-2b8ab99a-6470-11e8-9d62-dab0cf3b0d22.jpg)

#### Tell us about your browser and operating system

* Browser Name:  Firefox
* Browser Version:  60.0.1
* Operating System: Win10

**Comment by johnkennedy9147** [comment]
search is return in url: 
`https://learn.freecodecamp.org/coding-interview-questions-and-take-home-assignments/coding-interview-take-home-projects/build-a-wikipedia-viewer`

should be: 
`https://learn.freecodecamp.org/coding-interview-prep/take-home-projects/build-a-wikipedia-viewer`

Didnt check them all but seems it is all the projects in that section - path changed.

**Comment by Dimitrios51** [comment]
Yes this is the url that is returning at first but then, it returns another one with the 404 not found error. 
I tried the second url you sent and it is the right one, but I am not getting in when using the search box.

**Comment by jonathan-grah** [comment]
Yeah @johnkennedy9147 @Dimitrios51 , the path needs to be updated I believe.

**Comment by Dimitrios51** [comment]
Actually, others need to be updated too, as I tried searching for some other keywords too, and after clicking on the links I got again to the same problematic URL.

**Comment by hathway01** [comment]
can is work on this ?

**Comment by thecodingaviator** [comment]
@Bouncey Any clues as to where I can find this 'list' of search terms and fix it?

**Comment by theREALaltoid** [comment]
Working on this!

**Comment by moT01** [comment]
The issues I just closed appear to be the same problem as this.

To repeat what has been said above: The search results link to certain lessons/sections give a 404

Here is a list of what sections are problematic, and an example of the bad link from the search results, and what the actual url to the lesson is. 
- ALL RESPONSIVE WEB DESIGN PROJECTS:

Bad link - 
https://learn.freecodecamp.org/responsive-web-design/applied-responsive-web-design-projects/build-a-tribute-page
Actual URL -
https://learn.freecodecamp.org/responsive-web-design/responsive-web-design-projects/build-a-tribute-page/

- ALL APIS AND MICROSERVICES PROJECTS:

Bad link - 
https://learn.freecodecamp.org/apis-and-microservices/api-and-microservice-projects/timestamp-microservice
Actual URL -
https://learn.freecodecamp.org/apis-and-microservices/apis-and-microservices-projects/timestamp-microservice/

- ALL CODING INTERVIEW SECTIONS:

Bad algorithms link -
https://learn.freecodecamp.org/coding-interview-questions-and-take-home-assignments/coding-interview-algorithm-questions/find-the-symmetric-difference
Actual algorithms URL -
https://learn.freecodecamp.org/coding-interview-prep/algorithms/find-the-symmetric-difference

Bad data structures link -
https://learn.freecodecamp.org/coding-interview-questions-and-take-home-assignments/coding-interview-data-structure-questions/typed-arrays
Actual data structures URL -
https://learn.freecodecamp.org/coding-interview-prep/data-structures/typed-arrays/

Bad take home projects link -
https://learn.freecodecamp.org/coding-interview-questions-and-take-home-assignments/coding-interview-take-home-projects/build-a-wikipedia-viewer
Actual take home projects URL - 
https://learn.freecodecamp.org/coding-interview-prep/take-home-projects/build-a-wikipedia-viewer

Bad rosetta code link -
https://learn.freecodecamp.org/coding-interview-questions-and-take-home-assignments/rosetta-code-problems/100-doors
Actual rosette code URL -
https://learn.freecodecamp.org/coding-interview-prep/rosetta-code/100-doors/

Bad project euler link -
https://learn.freecodecamp.org/coding-interview-questions-and-take-home-assignments/project-euler-problems/problem-1-multiples-of-3-and-5
Actual project euler URL -
https://learn.freecodecamp.org/coding-interview-prep/project-euler/problem-1-multiples-of-3-and-5/

So basically the search links need to be updated to match the current structure of the lessons.

**Comment by moT01** [comment]
Hey @Bouncey, this issue and a few others can be fixed, I believe, by updating the search data in the `search-assets` repo - Since your the main contributor over there, you may need to do it, or point some of us in the right direction. It looks like there's some scripts over in that repo for updating data - perhaps they can just be ran, I'm not sure - I may be wrong, but after a little digging, it seems like that's where the problem lies. Just making sure you are aware of this issue.

**Comment by RandellDawson** [comment]
@raisedadead Any status on when this will be implemented?

**Comment by raisedadead** [comment]
> @raisedadead Any status on when this will be implemented?

When master is shipped.

**Comment by RandellDawson** [comment]
@raisedadead Since the search feature is changing, won't this be a won't fix or obsolete issue?
